controllers/config_controller.go

```diff
@@ -3,55 +3,99 @@ package controllers
 import (
 	"context"
 	"fmt"
+	"time"
 
-	storageoscomv1 "github.com/storageos/operator/api/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/util/workqueue"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	tkctrl "github.com/ondat/operator-toolkit/controller"
+	storageoscomv1 "github.com/storageos/operator/api/v1"
 )
 
 // ConfigReconciler reconciles StorageOS related configs.
 type ConfigReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
+
+	Namespace string
+	// ConfigEventSink allows to trigger reconciliation of the StorageOSCluster by its controller.
+	ConfigEventSink chan<- event.GenericEvent
 }
 
 // Reconcile changes of watched resources.
-func (r *ConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *ConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, reterr error) {
+	ctx, span, _, log := instrumentation.Start(ctx, "ConfigReconciler.Reconcile")
+	start := time.Now()
+	defer tkctrl.LogReconcileFinish(log, "reconciliation finished", start, &result, &reterr)
+	defer span.End()
+
+	stosList := storageoscomv1.StorageOSClusterList{}
+	err := r.Client.List(ctx, &stosList, &client.ListOptions{Namespace: r.Namespace})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	ok, stos := isMonitoredConfig(&stosList, &client.ObjectKey{Namespace: req.Namespace, Name: req.Name})
+	if !ok {
+		return ctrl.Result{}, nil
+	}
+
+	log.Info("monitored configmap has been changed", "req", req.String())
+	if stos.Labels == nil {
+		stos.Labels = make(map[string]string)
+	}
+	// trigger reconciliation for the StorageOSCluster
+	stos.Labels[triggerReconcileLabelKey] = triggerReconcileLabelValue
+	r.ConfigEventSink <- event.GenericEvent{Object: stos.DeepCopy()}
 	return ctrl.Result{}, nil
 }
 
+func isMonitoredConfig(
+	stosList *storageoscomv1.StorageOSClusterList,
+	obj *client.ObjectKey,
+) (bool, *storageoscomv1.StorageOSCluster) {
```

Comment:
If we rename the function something like `get...` we should remove the boolean return value, because `nil` also means false. WDYT?

Reply:
It makes for me :)